submissions/angels_unit_tests/angels_unit_tests.md

```diff
@@ -0,0 +1,55 @@
+
+## The ball sped up rolling downhill
+
+At the end of 2020, we decided to work on a big feature inside Angel's Industries: a decent implementation of the component and technology overhaul. This was the last feature Arch666Angel was working on before he pulled back from modding. On the 26th of May 2021 (half a year since the last release), we presented the new version of the overhauls to the public, after which the bug reports started rolling in quickly.
+
+We realised that with the overhauls we created many more configuration, with even more settings. At this point it becomes quite though to maintain and test this completely by hand. I started thinking about creating some [unit tests](https://en.wikipedia.org/wiki/Unit_testing) to take this burden of our shoulders. After some discussions and ideas from the great modding community I finally decided on the 30th of June 2021 to commit myself at creating the unit test infrastructure, groupted in a separate mod: [Angel's Dev - Unit Tests](https://github.com/Arch666Angel/mods/pull/634/commits/d59328e81235e26beb88ba66659cb78315827a00). A small note, You will not find this mod on the modportal.
+
+I already had a way of building the Angel's mods from the github folder to the factorio mod folder and a similar script to download bob mods from the mod portal. In order to test different configurations, I had to write two additional scripts. [The first script](https://github.com/Arch666Angel/mods/pull/634/commits/8977e17bba55d06fb3b60ff4c593e2307f110395) reads and rewrites the `mod-settings.dat` file with the mod settings. Since this was a part that I didn't encounter in creating mods before, I was very surprised at finding [thorough documentation on the Factorio Wiki](https://wiki.factorio.com/Mod_settings_file_format) about this. [The second script](https://github.com/Arch666Angel/mods/pull/634/commits/abb64c2fe8007e060f112716ebe33864510b6451) configures the `mod-list.json` file, which contains the information about which mods have to be enabled.
+
+With these 4 scripts, everything is in place to launch factorio with a certain configuration of the latest mods and their mod settings. It was only a matter of telling factorio to launch over and over with different mod settings. Assuming an ingame mod (Angel's Dev - Unit Tests in this case) telling the testing infrastructure that the testing is over, it can shut down Factorio and launch the next configuration to test.
```

Comment:
Maybe add a sentence a sentence as to what you achieved. Am I correct that this simply tested that all the mods load correctly - no matter what settings/mods. Basically crashes on startup, but not yet during runtime?

Reply:
Fixed in https://github.com/AlternativeFFFF/Alt-F4/pull/551/commits/0d2bc1d8be83f76d557f22d7d39ba53b96e0aa9f